model/shift_design_constraints.py

```diff
@@ -20,27 +20,40 @@ def __init__(
         self.long_shifts = shift_sets["long_shifts"]
 
         # Adding constraints
-        self.add_minimum_demand_coverage(var.x)
-        self.add_deviation_from_demand(var.x, var.delta)
+        self.add_minimum_demand_coverage(var.x, var.mu)
+        self.add_maximum_demand_coverage(var.mu)
+        self.add_deviation_from_demand(var.mu, var.delta)
         self.add_mapping_x_to_y(var.x, var.y)
         self.add_short_shift_duration(var.y, var.rho)
         self.add_long_shift_duration(var.y, var.rho)
 
     # Constraint definitions
-    def add_minimum_demand_coverage(self, x):
+    def add_minimum_demand_coverage(self, x, mu):
         self.model.addConstrs(
             (
                 quicksum(x[t_marked, v] for t_marked, v in self.shifts_overlapping_t[t])
-                >= quicksum(self.demand["min"][c, t] for c in self.competencies if self.demand["min"].get((c,t)))
+                == quicksum(self.demand["min"][c, t] for c in self.competencies if self.demand["min"].get((c, t)))
+                + quicksum(mu[c, t] for c in self.competencies)
                 for t in self.time_periods_combined
```

Comment:
I am not sure if this has the desired effect. This constraint states that
```
The sum of all allocated shifts == The sum of all demand + The sum of all MU
```

This allows for individual violations of the restriction, as long as the total deviation is within the limits. In ESP this would definitely be wrong. I am not sure if this applies to SDP as well, but I would imagine so.

Reply:
Since the MU is actually independent of _c_, the sum of all MU is actually just MU. So the expression should rally state:

`The sum of all allocated shifts == The sum of all demand + MU`

or in other words

`Number of "active" shifts at time t == Accumulated demand at time t + MU at time t`

I can´t see the point of individual violations? Could you explain more in detail?

Reply:
Given mininimum demand:
- `demand[c1, t1] = 2`
- `demand[c2, t1] = 2`

...and only one active shift:
- `x[t1, v1] = 4`

There is no way to violate the restriction with these values? 

Do I understand correctly if :
- `x` does not have any relation to `c`
- the only thing of interest is that there exists sufficient allocation of employees to shifts 
- whether there are sufficiently many employees to cover demand for different competencies are not relevant

If so, I think this is good - but there is still possible to improve upon this (to be more aligned with the intention of `mu`): 

```python
quicksum(
    x[t_marked, v] 
    for t_marked, v in self.shifts_overlapping_t[t]
)
== 
quicksum(
    quicksum(    
        self.demand["min"][c, t] 
        for c in self.competencies if self.demand["min"].get((c, t))
     )
     + mu[t]
     for t in self.time_periods_combined
)
```

What do you think? If you can confirm that my understanding is correct, I will approve the PR regardless of the outcome of this suggestion :)